### PR TITLE
Repl and cider improves

### PR DESCRIPTION
**Pull request by benjamin-asdf**
These fix some errors when connecting with cider nrepl. 
Also some improves where it loads default vars for repl and nrepl.
We also like to implement in cider skipping eval init code. 
(I'm working on a suggestion patch for cider at the moment).

### CODE REVIEW

**Comment by benjamin-asdf** [comment]
Roadmap:
1) modify cider so starting and connecting to nbb is smooth
2)? implement eldoc and such in the nrepl server

**Comment by borkdude** [comment]
Can you push another commit? I enabled building PRs on CircleCI.

**Comment by benjamin-asdf** [comment]
like an empty commit or something specific?

File: src/nbb/impl/repl.cljs [hunk 1]
```diff
@@ -170,6 +170,8 @@
   ([] (repl nil))
   ([_opts]
    (when tty (.setRawMode js/process.stdin true))
+   (doseq [form nbb/repl-requires]
```
Comment (borkdude) [inline src/nbb/impl/repl.cljs:173]: This should be implement async, because eval-require itself is async, similar to above.

**Comment by borkdude** [comment]
> like an empty commit or something specific?

IT's already building, so all good. I prefer no force-pushes in babashka and clj-kondo projects. A PR will be squashed anyway and it's easier to follow incremental updates.

File: src/nbb/impl/repl.cljs (hunk 1 above)
Comment (benjamin-asdf) [reply src/nbb/impl/repl.cljs:173]: you mean sync like with the .then chain right

Comment (benjamin-asdf) [reply src/nbb/impl/repl.cljs]: e8ced5773407813f922af5eb0349e9412ecd33a0

File: src/nbb/impl/nrepl_server.cljs [hunk 2]
```diff
@@ -227,5 +237,18 @@
...
               (fs/unlinkSync ".nrepl-port")))))
 
-(defn init []
-  (start-server @nbb/opts))
+(defn
+  init
+  []
+  (let [eval-require (fn
+                       [ns-form]
+                       (when ns-form
```
Comment (borkdude) [inline src/nbb/impl/nrepl_server.cljs:245]: why is there a `when` here? if there is no ns-form, then the `(.then nil ...)` expression will probably crash.

Comment (benjamin-asdf) [reply src/nbb/impl/nrepl_server.cljs]: Yea true. I removed the `when`. I also was thinking if this should be a function (in `core.cljs`?).
Like `(defn eval-repl-requires []) ; promise`

File: src/nbb/impl/repl.cljs [hunk 3]
```diff
@@ -170,8 +170,19 @@
   ([] (repl nil))
   ([_opts]
    (when tty (.setRawMode js/process.stdin true))
-   (js/Promise. (fn [resolve]
-                  (input-loop nil resolve)))))
+   (let [eval-require (fn
+                        [ns-form]
+                        (nbb/eval-require
+                         (list
+                          'quote
```
Comment (borkdude) [inline src/nbb/impl/repl.cljs:177]: The `'quote` originally is the `require` symbol in `(require '[....])` so it's probably clearer if we name that `require` here as well.

**Comment by borkdude** [comment]
I'll just go ahead and merge this. After that I'd like to work in accordance to the developer docs that I just updated:

https://github.com/babashka/nbb/blob/main/doc/dev.md

**Pull request merged**